### DISCUSSION TITLE
Modular support for different models

### DISCUSSION THREAD

**Discussion by KWKY7** [analyticsinmotion/add-chatgpt-to-microsoft-word#5] [Q&A] [answered]
are there any way to change the model from davinchi 003 to gpt3.5 turbo or even gpt4. Changed the strModel and even the selection fonts/colours but nothing has changed when reimport the bas file

**Comment by rossarmstrong** [comment, marked as answer]
Hi @kinenwong, thank you for your question.

**Quick answer**...we created an additional project so users could use the latest models. It can be found here: [https://github.com/analyticsinmotion/add-gpt-chatbot-to-microsoft-word](https://github.com/analyticsinmotion/add-gpt-chatbot-to-microsoft-word) 

If you already have this project set up, it should not take long at all to add the new one. You can run both projects simultaneously in Word with no problems and this also allows you to have access to both models for answer comparisons :)

**Longer answer**...the davinci-003 model uses a 'text' input/output type format. The gpt3.5 model uses a 'messaging' type format. The messaging format allows the later models to incorporate previous message history into its responses, thus providing a sort of pseudo context. The object for this is that the responses returned will hopefully be more relevant to what you are after.

As a result, the code to run the later models has to be structured differently to incorporate this capability. It made sense to create a separate project repository for GPT3.5.

Let us know how you go.

Cheers

**Reply by KWKY7** [reply]
thanks for the prompt response! This works as intended, but if i were to prefer gpt4, would the simple change be just to change the strModel to "gpt-4"?  

Edit: saw the comment about "limited beta". i guess i'll have to wait til further notice. Will keep a lookout on that page instead. Thanks!
